### PR TITLE
Setting up poetry

### PR DESCRIPTION
**Pull request by leikoilja**
- Setting up poetry
- Updating github actions to use poetry
- Updating contributions file
- Running pre-commit linters on all files

### CODE REVIEW

**Comment by KapJI** [comment]
I'm not sure if `tox` is still useful with poetry. Testing with different python version can be done using Github Actions and I think it's better just to replace it `pytest`.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[tool.poetry]
+name = "glocaltokens"
+version = "0.0.0"
```
Comment (KapJI) [inline pyproject.toml:3]: This should be an actual package version. Basically poetry should replace setuptools and `setup.py`.

Comment (KapJI) [reply pyproject.toml:3]: This can be done in a separate PR to split this migration in parts.

Comment (leikoilja) [reply pyproject.toml:3]: oh that's nice!

Comment (leikoilja) [reply pyproject.toml]: @KapJI I have just tried `poetry build` locally and it works flawlessly, super nice!

**Comment by leikoilja** [comment]
@KapJI, what you propose with `tox` sounds reasonable. But how would that look practically?
- Removing `tox` and `tox-gh-actions` from poetry
- Setting up github action to run tox independently from poetry?

**Comment by KapJI** [comment]
I mean removing `tox` at all and run `pytest` directly from `poetry` 🙂

File: pyproject.toml (hunk 1 above)
Comment (KapJI) [reply pyproject.toml]: Then you should be also able to use `poetry publish` to release the new version.

Comment (leikoilja) [reply pyproject.toml]: Exactly! Thanks for introducing `poetry`, such a time saver :P

**Comment by leikoilja** [comment]
Roger that. FIxed it and i like it now :) 
What do you think, @KapJI?
As next PR we can enable mypy on this repo i guess :)

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -28,6 +28,6 @@ jobs:
       - name: Install dependencies
         run: poetry install
 
-      - name: Run tox
+      - name: Run pytest
         run: |
-          poetry run tox
+          poetry run pytest
```
Comment (KapJI) [inline .github/workflows/tests.yaml]: Are're already running tests in linting job, this one can be removed.
You can rename linting to testing as well.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -16,3 +16,11 @@ repos:
         args:
           - --safe
           - --quiet
```
Comment (KapJI) [inline .pre-commit-config.yaml:18]: Replace `black` and `isort` entries with ones which run versions installed by poetry. Similar to `ha-google-home`.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+Activate `pre-commit` git hook:
+
+```console
+$ poetry run pre-commit install
+```
+
+Now the pre-commit tests will be done every time you commit.
+
+You can run the tests on all repository file with the command:
```
Comment (KapJI) [inline CONTRIBUTING.md:81]:
```suggestion
You can also run the tests on all repository files manually with the command:
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+pre-commit = "^2.11.1"
+pylint = "^2.7.2"
+isort = "^5.8.0"
+mypy = "^0.812"
+pytest = "^6.2.2"
+Faker = "^6.6.2"
+mock = "^4.0.3"
+
+[tool.poetry.urls]
+"Bug Tracker" = "https://github.com/leikoilja/glocaltokens/issues"
```
Comment (KapJI) [inline pyproject.toml:31]: Delete `HISTORY.md` and add link to release notes here.

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+[tool.poetry]
+name = "glocaltokens"
+version = "0.2.6"
+description = "Tool to extract Google device local authentication tokens in Python"
+authors = ["Ilja Leiko <leikoilja@gmail.com>"]
+license = "MIT"
+repository = "https://github.com/leikoilja/glocaltokens"
```
Comment (KapJI) [inline pyproject.toml:7]: Not sure if you need it but you can also add a `homepage` here.

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+readme = "README.md"
```
Comment (KapJI) [inline pyproject.toml:8]: You can add `keywords` in this section to match old `setup.py`. Although I'm not sure how useful they are.

Comment (KapJI) [reply pyproject.toml:8]: Also it's worth adding `classifiers` here. See [poetry documentation](https://python-poetry.org/docs/pyproject/) and [packaging documentation](https://packaging.python.org/guides/distributing-packages-using-setuptools/#classifiers).

**Review by KapJI — CHANGES_REQUESTED** [review]
Few final touches to fully deprecate `setup.py`.

File: pyproject.toml [hunk 8]
```diff
@@ -1,11 +1,24 @@
...
+keywords = [
+    "Authentication",
+    "Google Home",
+    "Google",
+    "LocalAuthenticationTokens",
+]
 license = "MIT"
-repository = "https://github.com/leikoilja/glocaltokens"
+name = "glocaltokens"
 readme = "README.md"
```
Comment (KapJI) [inline pyproject.toml]: Is it reordered by you or some tool? I think previous ordering was better with name and version on the top and links next to each other.

**Review by KapJI — APPROVED** [review]
Other than that, looks great!

File: tox.ini [hunk 9]
```diff
@@ -1,15 +1,15 @@
 [tox]
```
Comment (KapJI) [inline tox.ini:1]: Missed that. `tox.ini` should be removed as tox is no longer used.

File: pyproject.toml (hunk 8 above)
Comment (leikoilja) [reply pyproject.toml]: Just reordered it manually to be alphabetic :)

Comment (KapJI) [reply pyproject.toml]: I think this section is better to be semantically sorted 🙂

**Comment by leikoilja** [comment]
Awesome, thanks, @KapJI for help! 💣 Let's merge it in! 💥

**Pull request merged**